wasp/apps/musicplayer.py

```diff
@@ -0,0 +1,208 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# Copyright (C) 2020 Daniel Thompson
+# Copyright (C) 2020 Carlos Gil
+
+"""Music Player for GadgetBridge
+~~~~~~~~~~~~~
+
+Music Player Controller:
+- Touch: play/pause
+- Swipe UPDOWN: Volume down/up
+- Swipe LEFTRIGHT: next/previous
+"""
+
+import wasp
+
+import icons
+import time
+
+
+class MusicPlayerApp(object):
+    """ Music Player Controller application.
+
+    .. figure:: res/MusicPlayerApp.png
+        :width: 179
+
+        Screenshot of the Music Player application
+    """
+    NAME = 'MPlayer'
+    ICON = icons.headset
+
+    def __init__(self):
+
+        self._play_icon = icons.play
+        self._pause_icon = icons.pause
+        self._play_state = False
+        self._icon_state = self._play_icon
+        self._gb_play = '{"t":"music", "n":"play"} '
+        self._gb_pause = '{"t":"music", "n":"pause"} '
+        self._gb_next = '{"t":"music", "n":"next"} '
+        self._gb_prev = '{"t":"music", "n":"previous"} '
+        self._gb_vup = '{"t":"music", "n":"volumeup"} '
+        self._gb_vdo = '{"t":"music", "n":"volumedown"} '
+        self._max_display = 240
+        self._icon_size = 72
+        self._center_at = int((self._max_display - self._icon_size)/2)
+        self._musicstate = 'pause'
+        self._artist = ''
+        self._track = ''
+        self._state_changed = True
+        self._track_changed = True
+        self._artist_changed = True
+        self._track_pos = 24 + 144
+        self._artist_pos = 12
+
```

Comment:
This seems like a very large number of class variables. I'm rather worried about the RAM usage.

If it works there's no need to worry too much... but I'll be inclined to add this to the (optional_ Application Library rather than enabling it by default for the time being.